worker/solution.go

```diff
@@ -1,43 +1,80 @@
 package worker
-
 import (
-	"github.com/mini-maxit/worker/models"
 	"gorm.io/gorm"
 )
 
 
+type TaskForRunner struct {
+	BaseDir            string
+	LanguageType       string
+	LanguageVersion    string
+	SolutionFileName   string
+	StdinDir           string
+	ExpectedOutputsDir string
+	TimeLimits 	       []float64
+	MemoryLimits 	   []float64
+}
+
+type InputOutputData struct {
+	TimeLimit  		float64  `gorm:"column:time_limit"`
+	MemoryLimit     float64  `gorm:"column:memory_limit"`
+	Order 		    int      `gorm:"column:order"`
+}
+
+type SolutionConfig struct {
+	LanguageType    string  `gorm:"column:language_type"`
+	LanguageVersion string  `gorm:"column:language_version"`
+	SolutionFileName string `gorm:"column:solution_file_name"`
+}
+
+type DirConfig struct {
+    BaseDir            string `gorm:"column:dir_path"`
+    StdinDir           string `gorm:"column:input_dir_path"`
+    ExpectedOutputsDir string `gorm:"column:output_dir_path"`
+}
+
+
+
 // GetDataForSolutionRunner retrieves the data needed to run the solution
-func getDataForSolutionRunner(db *gorm.DB, task_id, user_solution_id, input_output_id int) (models.Task, error) {
-	var task models.Task
+func getDataForSolutionRunner(db *gorm.DB, task_id int, user_solution_id int, input_output_ids []int) (TaskForRunner, error) {
+	var task TaskForRunner
+
+	var dirConfig DirConfig
 
-    // Get the base directory path for the task
-	err := db.Table("tasks").Select("dir_path").Where("id = ?", task_id).Scan(&task.BaseDir).Error
+    // Get the directories configuration - base dir, input dir, output dir
+	err := db.Table("tasks").Select("dir_path, input_dir_path, output_dir_path").Where("id = ?", task_id).Scan(&dirConfig).Error
 	if err != nil {
-		return models.Task{}, err
+		return TaskForRunner{}, err
 	}
 
+	task.BaseDir = dirConfig.BaseDir
```

Comment:
Remember that the dir in database corresponds to directory in FileStorage, the worker should download a copy of this directory and use local paths in all of it services. So you cannot just assign base dir to to task